apps/explorer/src/components/OwnedObjectsV2/OwnedObjects.tsx

```diff
@@ -40,32 +32,32 @@ function OwnerCoins({ id }: { id: string }): JSX.Element {
                     <Heading color="gray-90" variant="heading4/semibold">
                         NFTs
                     </Heading>
-
-                    <div className="flex max-h-80 flex-col overflow-auto">
-                        <div className="flex flex-wrap">
-                            {results
-                                .slice(
-                                    (currentSlice - 1) * OBJECTS_PER_PAGE,
-                                    currentSlice * OBJECTS_PER_PAGE
-                                )
-                                .map((obj) => (
+                    {isLoading || isFetching ? (
+                        <LoadingSpinner />
+                    ) : (
+                        <div className="flex max-h-80 flex-col overflow-auto">
+                            <div className="flex flex-wrap">
+                                {data?.pages[currentPage].data.map((obj) => (
```

Comment:
I think the `data` part should be fine since `isLoading` should handle it. I was wondering about the indexing of the `pages` array. Can that be falsy?

ie. `data.pages[currentPage]?.data.map` for safety

Reply:
I'll add it in regardless! I think if that happens there would likely an error thrown by the api of some sort which is being caught so I wouldn't expect that to be an issue